src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { Filter } from '../../utils/Filter';
+
+type Props = {
+  visibleTodos: Todo[];
+  isTodoEditing: boolean;
+  selectedPostId: number;
+  setIsTodoEditing: React.Dispatch<React.SetStateAction<boolean>>;
+  setSelectedPostId: React.Dispatch<React.SetStateAction<number>>;
+  selectedFilter: Filter;
+  onDelete: (todoId: number) => Promise<void>;
+  onUpdate: (todo: Todo) => Promise<void>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleTodos,
+  isTodoEditing,
+  selectedPostId,
+  setIsTodoEditing,
+  setSelectedPostId,
+  selectedFilter,
+  onDelete,
+  onUpdate,
+}) => {
+  let todosCopy: Todo[];
+
+  switch (selectedFilter) {
+    case Filter.all:
+      todosCopy = [...visibleTodos];
+      break;
+
+    case Filter.active:
+      todosCopy = [...visibleTodos].filter(todo => !todo.completed);
```

Comment:
```suggestion
      todosCopy = visibleTodos.filter(todo => !todo.completed);
```
the filter method doesn't mutate the array